app/models.py

```diff
@@ -0,0 +1,195 @@
+                algorithm='HS256'
+                )
+            return jwt_string
+        except Exception as e:
+            return str(e)
+
+    def save(self):
+        """Method saves user to the database"""
+        db.session.add(self)
+        db.session.commit()
+
+    def delete(self):
+        """Method deletes user from database"""
+        db.session.delete(self)
+        db.session.commit()
+
+    def __repr__(self):
+        """ """
+        return '<User %r>' %(self.name)
+
+    @staticmethod
+    def decode_auth_token(token):
+        """Method decodes authentication token"""
+        try:
+            payload = jwt.decode(
+                token,
+                current_app.config.get('SECRET')
+                )
+            token_blacklisted = BlacklistToken.blacklisted(token)
+            if token_blacklisted:
+                return 'Token no longer valid login again'
+            return payload['sub']
+        except jwt.ExpiredSignatureError:
+            return 'Expired token'
+        except jwt.InvalidTokenError:
+            return 'Invalid token'
+
+class Bucketlist(db.Model):
+    """Class to define the bucketlists table"""
+    __tablename__ = 'bucketlists'
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(256), nullable=False, unique=True)
+    description = db.Column(db.Text)
+    owner = db.Column(db.Integer, db.ForeignKey(User.id))
+    items = db.relationship('Item', backref='Bucketlist')
+
+    def __init__(self, name, description, owner):
+        """Initialising the bucketlist"""
+        self.name = name
+        self.description = description
+        self.owner = owner
+
+    def save(self):
+        """Method to save bucketlist to database"""
+        db.session.add(self)
+        db.session.commit()
+
+    def delete(self):
+        """Method to delete bucketlist from database"""
+        db.session.delete(self)
+        db.session.commit()
+
+    def to_json(self):
+        """Method converts a bucketlist to a json object"""
+        json_data = {
+            'id':self.id,
+            'name': self.name,
+            'description': self.description,
+            'owner': self.owner,
+            'items': [item.to_json() for item in self.items]
+        }
+        return json_data
+
+    @staticmethod
+    def get_all_bucketlists(owner_id):
+        """Method returns all bucketlists owned by a given user"""
+        return Bucketlist.query.filter_by(owner=owner_id)
+
+    def __repr__(self):
+        """A representation for an instance of a bucketlist"""
+        return '<Bucketlist: %r>' %(self.name)
+
+class Item(db.Model):
+    """Class to define the Items table"""
+    __tablename__ = 'items'
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(256), nullable=False, unique=True)
+    description = db.Column(db.Text)
+    bucketlist_id = db.Column(db.Integer, db.ForeignKey(Bucketlist.id))
```

Comment:
You might need a cascade delete property to automatically delete item when bucket is deleted